public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -410,11 +412,12 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
       end: 0,
     };
     const range = Math.ceil(end - start);
-
+    // target.stepOption specifies whether to use min, max or exact step
+    const stepOption = (target.stepOption || DEFAULT_STEP_OPTION.value) as StepType;
```

Comment:
kind of a nit here, i think this should work as well:
```suggestion
    const stepOption: StepType = target.stepOption || DEFAULT_STEP_OPTION.value;
```

the reason is: 
```ts
type Num = 'one' | 'two' | 'three'

const a: Num = 'one';

const b: Num = 'tree' as Num // this won't cause any error, but vegetables are not valid numbers
```

Reply:
Yeah that would be better! But when i try it i'm getting this error 
```
Type 'StepType | undefined' is not assignable to type 'StepType'.
  Type 'undefined' is not assignable to type 'StepType'.
```
I don't know what i should change to fix this :(

Reply:
I changed it to `const stepOption = target.stepOption || DEFAULT_STEP_OPTION.value as StepType;` so that it type casts the default step option because that will always be of type StepType. What do you think?